modules/personalize_taxonomy_context/personalize_taxonomy_context.module

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * @file personalize_taxonomy_context.module
+ * Provides a visitor context plugin for taxonomy.
+ */
+
+/**
+ * Implements hook_personalize_visitor_contexts().
+ */
+function personalize_taxonomy_context_personalize_visitor_context() {
+  $info = array();
+  $path = drupal_get_path('module', 'personalize_taxonomy_context') . '/plugins';
+  $info['taxonomy_context'] = array(
+    'path' => $path . '/visitor_context',
+    'handler' => array(
+      'file' => 'TaxonomyContext.inc',
+      'class' => 'TaxonomyContext',
+    ),
+  );
+  return $info;
+}
+
+/**
+ * Implements hook_node_view().
+ */
+function personalize_taxonomy_context_node_view($node, $view_mode, $langcode) {
+  personalize_taxonomy_context_node_taxonomy($node, $view_mode == 'full');
+}
+
+/**
+ * Handles the node-specific taxonomy data.
+ *
+ * @param $node
+ *   (optional) If passed then the taxonomy terms for the selected node
```

Comment:
Same reason you did in [here](https://github.com/acquia/visitor_actions/blame/7.x-1.x/visitor_actions.module#L617) and [here](https://github.com/acquia/acquia_lift/blame/7.x-1.x/acquia_lift_profiles/acquia_lift_profiles.module#L433)? =D

To be serious, I think it is "optional" because 1) when the node is passed in, we are scooping up the node's taxonomy, and 2) when the node is not passed in, we are using the function to retrieve statically stored taxonomy.